modules/k8s/secret_test.go

```diff
@@ -40,6 +40,18 @@ func TestGetSecretEReturnsCorrectSecretInCorrectNamespace(t *testing.T) {
 	require.Equal(t, secret.Namespace, uniqueID)
 }
 
+func TestWaitUntilSecretAvailableReturnsSuccessfully(t *testing.T) {
+	t.Parallel()
+
+	uniqueID := strings.ToLower(random.UniqueId())
+	options := NewKubectlOptions("", "", uniqueID)
+	configData := fmt.Sprintf(EXAMPLE_SECRET_YAML_TEMPLATE, uniqueID, uniqueID)
+	defer KubectlDeleteFromString(t, options, configData)
+
+	KubectlApplyFromString(t, options, configData)
+	WaitUntilServiceAvailable(t, options, uniqueID, 10, 1*time.Second)
```

Comment:
Should this be `WaitUntilSecretAvailable`?

Reply:
And should we try to read the secret after?

Reply:
Ah, yes, it should be! I think I must be having issues running the tests locally. 

I can read the secret if you'd like, but that functionality is covered by the GetSecretE test. I tried to stick to the testing convention used in the WaitUntilServiceAvailable test in service_test.go. Let me know what you think. If you'd be more comfortable with reading the secret after it comes available, I'll throw it in there!